shared/js/l10n.js

```diff
@@ -1607,27 +1684,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this, meta));
+    } else {
+      registerLocales.call(this, meta);
     }
-    return initLocale.call(this);
```

Comment:
well, no. Because in case of LangpackAPI use we are becoming asynchronous so initLocale must wait for registerLocales which waits to getAdditionalLanguages.

Removing the new registerLocales will mean that all its code (with the stuff for merging available and extra langs) will be moved to initResources growing this function a lot. Do you want it?